src/node_debug_options.cc

```diff
@@ -0,0 +1,133 @@
+#include "node_debug_options.h"
+
+#include <string.h>
+#include "util.h"
+
+namespace node {
+
+namespace {
+const int default_debugger_port = 5858;
+const int default_inspector_port = 9229;
+
+}  // namespace
+
+DebugOptions::DebugOptions() : debug_enabled_(false),
+#if HAVE_INSPECTOR
+                               inspector_enabled_(false),
+#endif  // HAVE_INSPECTOR
+                               wait_connect_(false), http_enabled_(false),
+                               host_name_("127.0.0.1"), port_(-1) { }
+
+void DebugOptions::EnableDebugAgent(DebugAgentType tool) {
+  switch (tool) {
+#if HAVE_INSPECTOR
+    case DebugAgentType::kInspector:
+      inspector_enabled_ = true;
+      debug_enabled_ = true;
+      break;
+#endif  // HAVE_INSPECTOR
+    case DebugAgentType::kDebugger:
+      debug_enabled_ = true;
+      break;
+    case DebugAgentType::kNone:
+      break;
+  }
+}
+
+bool DebugOptions::ParseOption(const std::string& option) {
+  bool enable_inspector = false;
+  bool has_argument = false;
+  std::string option_name;
+  std::string port;
+
+  auto pos = option.find("=");
+  if (pos != std::string::npos) {
+    has_argument = true;
+    option_name = option.substr(0, pos);
+    port = option.substr(pos + 1);
+  } else {
+    option_name = option;
+  }
+
+  if (option_name == "--debug") {
+    debug_enabled_ = true;
+  } else if (option_name == "--debug-brk") {
+    debug_enabled_ = true;
+    wait_connect_ = true;
+  } else if (option_name == "--debug-port" && has_argument) {
+    // Do nothing - argument will be parsed below
+  // Specifying both --inspect and --debug means debugging is on, using Chromium
+  // inspector.
+  } else if (option_name == "--inspect") {
+    debug_enabled_ = true;
+    enable_inspector = true;
+  } else {
+    return false;
+  }
+
+  if (enable_inspector) {
+#if HAVE_INSPECTOR
+    inspector_enabled_ = true;
+#else
+    fprintf(stderr,
+            "Inspector support is not available with this Node.js build\n");
+    return false;
+#endif
+  }
+
+  if (!has_argument) {
+    return true;
+  }
+
+  // FIXME(bnoordhuis) Move IPv6 address parsing logic to lib/net.js.
+  // It seems reasonable to support [address]:port notation
+  // in net.Server#listen() and net.Socket#connect().
+  const size_t port_len = port.size();
+  if (port[0] == '[' && port[port_len - 1] == ']') {
+    host_name_ = port.substr(1, port_len - 2);
+    return true;
+  }
+
+  auto colon = port.rfind(':');
+  if (colon == std::string::npos) {
+    // Either a port number or a host name.  Assume that
+    // if it's not all decimal digits, it's a host name.
+    for (char c : port) {
+      if (c < '0' || c > '9') {
+        host_name_ = port;
+        return true;
+      }
+    }
+  } else {
+    const int skip =
+        (colon > 0 && port[0] == '[' && port[colon - 1] == ']') ? 1 : 0;
```

Comment:
The ternary is superfluous here, bools decay to ints, and it makes it harder to read, IMO.